src/parse/ast.rs

```diff
@@ -1,11 +1,143 @@
-//! Defines types of the abstract syntax tree
-//!
-
 /// Top level type. Is returned by `parse`.
 #[derive(Debug, Clone)]
 pub enum Query {
-    Select,
-    Insert,
-    Update,
-    Delete
+    DefStmt(DefStmt),
+    ManipulationStmt(ManipulationStmt)
+}
```

Comment:
Like I already said to your team in person: I would flatten the tree by removing the distinction between `DefStmt` and `ManipulationStmt`. I would insert all enum variants of both of theses variants into the `Query` enum. Means: `Query` enum looks a bit like:
```rust
enum Query { Create, Alter, Drop, Update, Select, Insert, Delete }
```
It will avoid some ugly code in the parser and the Query Execution Unit, but as I already said: It's your choice ;)